packages/ove-app-networks/src/client/constants/networks.js

```diff
@@ -4,11 +4,63 @@ const Constants = {
     **************************************************************/
     DEFAULT_STATE_NAME: 'SigmaSample',
 
+    /**************************************************************
+                               Back-end
+    **************************************************************/
+    SOCKET_REFRESH_DELAY: 5000, // Unit: milliseconds
+    SOCKET_READY_WAIT_TIME: 3000, // Unit: milliseconds
+    HTTP_HEADER_CONTENT_TYPE: 'Content-Type',
+    HTTP_CONTENT_TYPE_JSON: 'application/json',
+
     /**************************************************************
                                 Common
     **************************************************************/
     CONTENT_DIV: '#graphArea',
     APP_NAME: 'networks'
 };
 
+/**************************************************************
+                            Enums
+**************************************************************/
+Constants.Operation = {
+    SEARCH: 'search',
+    COLOR: 'color',
+    LABEL: 'labelNodes',
+    NEIGHBORS_OF: 'neighborsOf'
+};
+
+Constants.Evaluation = {
+    PROPERTY: 'property',
+    LITERAL: 'literal',
+    FUNCTION_CALL: 'functionCall'.toLowerCase(),
+    EQUALS: 'eq',
+    NOT_EQUALS: 'ne',
+    LESS_THAN: 'lt',
+    GREATER_THAN: 'gt',
+    LESS_THAN_OR_EQUALS: 'le',
+    GREATER_THAN_OR_EQUALS: 'ge',
+    AND: 'and',
+    OR: 'or',
+    ADD: 'add',
+    SUBTRACT: 'sub',
+    MULTIPLY: 'mul',
+    DIVIDE: 'div',
+    MODULO: 'mod',
+
+    // The OData specification requires lower-case function names.
+    Function: {
+        SUBSTRING: 'substring',
+        SUBSTRING_OF: 'substringOf'.toLowerCase(),
```

Comment:
why not remove the toLowerCase() call in order to improve performance?

Reply:
There is really no performance difference since these are constants. Not having the function called explicitly did make linters and spellcheckers call these out and potentially introduce a mistaken fix (could even be automatic in some tools). The spec is very strict on the case of these names, and therefore it will instantly break the logic. This is why I included the `toLowerCase` function that makes the code much more resilient and readable.